### PR TITLE
NDEV-985 neon-cli add data source

### PR DESCRIPTION
**Pull request by valiksinev**
_Solana rpc-client:_
neon-cli was tested with proxy:
https://github.com/neonlabsorg/proxy-model.py/actions/runs/3577129534/jobs/6015772508

_Postgress client:_
Neon-cli  was tested by hands with tracer-api infrastructure.
Did not manage to test 'emulate' command, because of the  EVM_LOADER account  is absent in database.
Only the 'get-ether-account-data' command was successfully tested.

### CODE REVIEW

File: evm_loader/cli/src/main.rs [hunk 1]
```diff
@@ -704,8 +721,28 @@ fn main() {
             true,
         ).ok();
 
+
+        let rpc_client = if let Some(slot) = app_matches.value_of("slot") {
+            let slot:u64 = slot.parse().unwrap();
```
Comment (Deniskore) [inline evm_loader/cli/src/main.rs:726]: Please handle it properly without unwrap.

File: evm_loader/cli/src/main.rs [hunk 2]
```diff
@@ -704,8 +721,28 @@ fn main() {
...
+
+            let db_config = app_matches.value_of("db_config")
+                .map(|path|{ solana_cli_config::load_config_file(path).unwrap()})
```
Comment (Deniskore) [inline evm_loader/cli/src/main.rs:729]: Please handle it properly without unwrap.

File: evm_loader/cli/src/main.rs [hunk 3]
```diff
@@ -704,8 +721,28 @@ fn main() {
...
+                .unwrap();
```
Comment (Deniskore) [inline evm_loader/cli/src/main.rs:730]: Please handle it properly without unwrap.

File: evm_loader/cli/src/rpc/db.rs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+impl PostgresClient {
+    pub fn new(config: &DBConfig, slot: u64) -> Self {
+        let connection_str= format!("host={} port={} dbname={} user={} password={}",
+                                    config.host, config.port, config.database, config.user, config.password);
+
+        let (client, connection) = block(|| async {
+            connect(&connection_str, NoTls).await
+        }).unwrap();
```
Comment (Deniskore) [inline evm_loader/cli/src/rpc/db.rs:45]: Please handle it properly without unwrap.
You should also try to reconnect several times, not just one, and after that display an error message if something is wrong.

File: evm_loader/cli/src/rpc/db.rs [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+        if rows.len() != 1 {
+            return Ok(None);
+        }
+
+        let row = &rows[0];
+        let lamports: i64 = row.try_get(2)?;
+        let rent_epoch: i64 = row.try_get(4)?;
+        Ok(Some(Account {
+            lamports: u64::try_from(lamports).unwrap(),
```
Comment (Deniskore) [inline evm_loader/cli/src/rpc/db.rs:72]: Please handle it properly without unwrap.

File: evm_loader/cli/src/rpc/db.rs [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+            data: row.try_get(5)?,
+            owner: Pubkey::new(row.try_get(1)?),
+            executable: row.try_get(3)?,
+            rent_epoch: u64::try_from(rent_epoch).unwrap(),
```
Comment (Deniskore) [inline evm_loader/cli/src/rpc/db.rs:76]: Please handle it properly without unwrap.

File: evm_loader/cli/src/rpc/db.rs [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+        Ok(time)
+    }
+
+    pub fn get_latest_blockhash(&self) -> Result<String, Error> {
+        let slot: i64 = block(|| async {
+            self.client.query_one("SELECT MAX(slot) FROM public.slot", &[])
+                .await
+        })?.try_get(0)?;
+
+        self.get_block_hash(u64::try_from(slot).unwrap())
```
Comment (Deniskore) [inline evm_loader/cli/src/rpc/db.rs:108]: Please handle it properly without unwrap.

**Review by Deniskore — CHANGES_REQUESTED** [review]
We should discuss the possibility of using [async_trait](https://crates.io/crates/async-trait) to improve code readability where you use block() function

File: evm_loader/cli/src/rpc/db.rs (hunk 4 above)
Comment (valiksinev) [reply evm_loader/cli/src/rpc/db.rs]: fixed

Comment (valiksinev) [reply evm_loader/cli/src/rpc/db.rs]: fixed

File: evm_loader/cli/src/main.rs (hunk 1 above)
Comment (valiksinev) [reply evm_loader/cli/src/main.rs:726]: It is necessary to replace all unwrap() call in the application.
I will do it in next task

**Review by ivandzen — DISMISSED** [review]

**Review by anton-lisanin — APPROVED** [review]

**Review by Deniskore — APPROVED** [review]

**Pull request merged**